### PR TITLE
feat: implement RuntimeConfig

### PR DESCRIPTION
**Pull request by popomore**
RuntimeConfig can be injected in tegg and standalone

It contains

- baseDir
- env
- name

##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: core/common-util/src/RuntimeConfigUtil.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+   */
+  env: EnvType;
+
+  /**
+   * Application directory
+   */
+  baseDir: string;
+}
+
+export class RuntimeConfigUtil {
```
Comment (killagu) [inline core/common-util/src/RuntimeConfigUtil.ts:21]: 这里应该只是一个定义
```ts
class RuntimeConfig {
  name: string;
  env: string;
  baseDir: string;
}
```

在具体的 runtime 中把这个类构造出来。

这样就能解决多个实例的问题了。

Comment (popomore) [reply core/common-util/src/RuntimeConfigUtil.ts:21]: 所以这里还是类型注入的模式？用法就是 

```typescript
import { RuntimeConfig } from '@alipay/tegg-common-util';

@SingletonProto()
class Example {
  @Inject()
  runtimeConfig: RuntimeConfig;

}
```

**Review by killagu — APPROVED** [review]
LGTM

File: plugin/tegg/app.ts [hunk 2]
```diff
@@ -20,6 +21,11 @@ export default class App {
   }
 
   configWillLoad() {
+    RuntimeConfigUtil.setRuntimeConfig({
```
Comment (fengmk2) [inline plugin/tegg/app.ts:24]: 这里也改成一下就好了

**Comment by killagu** [comment]
LGTM

**Review by fengmk2 — APPROVED** [review]

**Pull request merged**